src/chapterintrofx.c

```diff
@@ -0,0 +1,1344 @@
+
+    CopyDataWithPossibleUncomp(gUnknown_08B196D8, gUnknown_02020188);
+    CallARM_FillTileRect(gBG2TilemapBuffer, gUnknown_02020188, 0x4000);
+
+    BG_EnableSyncByMask(4);
+
+    return;
+}
+
+void ChapterIntro_80209D8(struct ChapterIntroFXProc* proc) {
+    SetSpecialColorEffectsParameters(1, proc->unk_4C, 0x10, 0);
+
+    if ((proc->unk_50 == 3) || ((GetGameClock() & 3) == 0)) {
+        if (proc->unk_52 != 0) {
+            proc->unk_4C += 4;
+        } else {
+            proc->unk_4C++;
+        }
+
+        if (proc->unk_4C > 5) {
+            SetSpecialColorEffectsParameters(1, 6, 0x10, 0);
+            Proc_Break(proc);
+        }
+    }
+
+    return;
+}
+
+void ChapterIntro_8020A40(struct ChapterIntroFXProc* proc) {
+    proc->unk_4C = 0x1E;
+
+    sub_8001710();
+
+    sub_800172C(gUnknown_02022928, 4, 2, -1);
+    sub_800172C(gUnknown_02022928 + 0xA0, 0xE, 2, -1);
+    sub_800172C(gUnknown_02022928 + 0xE0, 0x12, 1, -1);
+
+    return;
+}
+
+void ChapterIntro_8020A8C(struct ChapterIntroFXProc* proc) {
+    int clock = GetGameClock() & 3;
+
+    if ((clock) == 0) {
+        sub_80D74B0();
+        EnablePaletteSync();
+
+        proc->unk_4C--;
+
+        if (proc->unk_4C < 0) {
+            gLCDControlBuffer.dispcnt.bg0_on = 1;
+            gLCDControlBuffer.dispcnt.bg1_on = 0;
+            gLCDControlBuffer.dispcnt.bg2_on = 0;
+            gLCDControlBuffer.dispcnt.bg3_on = 0;
+            gLCDControlBuffer.dispcnt.obj_on = 1;
+
+            SetBackgroundTileDataOffset(2, 0);
+            gPaletteBuffer[0] = clock;
+            EnablePaletteSync();
+            Proc_Break(proc);
+        }
+    }
+
+    return;
+}
+
+void ChapterIntro_8020AF8() {
+    SetupBackgrounds(0);
+    sub_80156D4();
+    AllocWeatherParticles(gRAMChapterData.chapterWeatherId);
+    SMS_UpdateFromGameData();
+    SMS_FlushIndirect();
+    Font_LoadForUI();
+
+    return;
+}
+
+void ChapterIntro_8020B20() {
+    gUnknown_0202BCB0.camera.y = 0xA0 * 4;
+
+    return;
+}
+
+#if NONMATCHING
+
+void ChapterIntro_InitMapDisplay() {
+    u16 var;
+
+    gLCDControlBuffer.dispcnt.bg0_on = 1;
+    gLCDControlBuffer.dispcnt.bg1_on = 1;
+    gLCDControlBuffer.dispcnt.bg2_on = 1;
+    gLCDControlBuffer.dispcnt.bg3_on = 1;
+    gLCDControlBuffer.dispcnt.obj_on = 1;
+
+    SetSpecialColorEffectsParameters(1, 0x10, 0, 0);
+
+    sub_8001ED0(0, 1, 0, 0, 0);
+    sub_8001F0C(0, 0, 0, 1, 1);
+
+    sub_8001F64(1);
+
+    CpuFastFill(0, (void *)BG_VRAM, 32);
+
+    BG_Fill(gBG0TilemapBuffer, 0);
+    BG_Fill(gBG1TilemapBuffer, 0);
+    BG_Fill(gBG2TilemapBuffer, 0);
+
+    BG_EnableSyncByMask(7);
+
+    DisableMapPaletteAnimations();
+
+    UnpackChapterMapGraphics(gRAMChapterData.chapterIndex);
+
+    SetupMapSpritesPalettes();
+    LoadObjUIGfx();
+
+    // TODO - Nonmatching is caused by an extra use of r2 around here for the results of this operation
+
+    var = GetROMChapterStruct(gRAMChapterData.chapterIndex)->_unk10[2];
```

Comment:
did you try encapsulating these areas in do{}while(0)s and locally initializing var in each scope?